jbpm-document/src/main/java/org/jbpm/document/service/impl/DocumentImpl.java

```diff
@@ -42,24 +42,36 @@
     private Map<String, String> attributes;
 
     public DocumentImpl() {
+        // Setting default values for identifier && download link
+        this.identifier = UUID.randomUUID().toString();
     }
 
-    public DocumentImpl(String identifier, String name, long size, Date lastModified) {
+    public DocumentImpl(String identifier,
+                        String name,
+                        long size,
+                        Date lastModified) {
         this.identifier = identifier;
         this.name = name;
         this.size = size;
         this.lastModified = lastModified;
         attributes = new HashMap<String, String>();
     }
 
-    public DocumentImpl(String name, long size, Date lastModified) {
+    public DocumentImpl(String name,
```

Comment:
shouldn't this constructor set the identifier to generated on?

Reply:
@mswiderski it's calling the deafault constructor this(), which already does it.

Reply:
damn it, I missed it, then it's all good, thanks :)